### ISSUE TITLE
Unable to change pod labels

### ISSUE THREAD

**Issue by FlorinAlexandru** [kubernetes-client/java#958]
I want to change (edit/add) labels on pods with specific labels. I have the following code:

    public void changeLabelsForPodFilterByLabels(String namespace, Map<String, String> filterByLabels, Map<String, String> newLabels) {
        try {
            V1PodList podList = coreV1Api.listNamespacedPod(namespace, null, null, null, null, null, null, null, false);
            List<V1Pod> podsFilteredByLabel = filterPodsByLabels(podList.getItems(), filterByLabels);
            List<V1Pod> labelModifiedPods = addLabels(podsFilteredByLabel, newLabels);
            List<V1Pod> updatedPods = updatePods(namespace, labelModifiedPods);
        } catch (ApiException e) {
            logger.error("API Exception", e);
            logger.error(e.getMessage());
            logger.error(e.getResponseBody());
        }
    }

    private List<V1Pod> filterPodsByLabels(List<V1Pod> pods, Map<String, String> labels) {
        List<V1Pod> filteredPods = new ArrayList<>();
        logger.info("Input labels: " + labels.toString());
        for(V1Pod pod: pods) {
            logger.info("Pod: " + pod.getMetadata().getName());
            Map<String, String> podLabels = pod.getMetadata().getLabels();
            logger.info("Labels:" + podLabels.toString());
            if (podLabels.entrySet().containsAll(labels.entrySet())) {
                filteredPods.add(pod);
            }
        }
        return filteredPods;
    }

    private List<V1Pod> addLabels(List<V1Pod> pods, Map<String, String> newLabels) {
        List<V1Pod> updatedPods = new ArrayList<>();
        for (V1Pod pod: pods) {
            pod.getMetadata().setLabels(newLabels);
            updatedPods.add(pod);
        }
        return updatedPods;
    }

    private List<V1Pod> updatePods(String namespace, List<V1Pod> updatedPods) throws ApiException {

        for (V1Pod pod: updatedPods){
            String body = generateBody("demo", "myvalue");
            V1Patch patch = new V1Patch(body);
            coreV1Api.patchNamespacedPod(pod.getMetadata().getName(), namespace, patch, null, null, null, false);
        }
        return null;
    }

    public String generateBody(String labelKey, String newLabelValue){
        return String.format("[{\"op\": \"add\", \"path\": \"/metadata/labels/%s\", \"value\": \"%s\"}]", labelKey, newLabelValue);
    }

I'm using version 8.0.2 of the Java library and Kubernetes 1.18.

Based on the example from here and from this thread and this thread I understand that a body should be provided as parameter to _coreV1Api.patchNamespacedPod_ method.

Can someone provide an example on how a body should look like? Currently I receive  the following error: `io.kubernetes.client.openapi.ApiException: Unprocessable Entity`

Debug log:

    2020-05-25 11:13:11.812  INFO 1 --- [nio-8080-exec-1] okhttp3.OkHttpClient                     : [{"op": "add", "path": "/metadata/labels/demo", "value": "myvalue"}]
    2020-05-25 11:13:11.812  INFO 1 --- [nio-8080-exec-1] okhttp3.OkHttpClient                     : --> END PATCH (68-byte body)
    2020-05-25 11:13:11.816  INFO 1 --- [nio-8080-exec-1] okhttp3.OkHttpClient                     : <-- 422 Unprocessable Entity https://10.96.0.1/api/v1/namespaces/default/pods/nginx-deployment-zone2-75fbb7d6d8-4td27?force=false (4ms)
    2020-05-25 11:13:11.817  INFO 1 --- [nio-8080-exec-1] okhttp3.OkHttpClient                     : Content-Type: application/json
    2020-05-25 11:13:11.817  INFO 1 --- [nio-8080-exec-1] okhttp3.OkHttpClient                     : Date: Mon, 25 May 2020 11:13:11 GMT
    2020-05-25 11:13:11.817  INFO 1 --- [nio-8080-exec-1] okhttp3.OkHttpClient                     : Content-Length: 392
    2020-05-25 11:13:11.817  INFO 1 --- [nio-8080-exec-1] okhttp3.OkHttpClient                     :
    2020-05-25 11:13:11.817  INFO 1 --- [nio-8080-exec-1] okhttp3.OkHttpClient                     : {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"PatchOptions.meta.k8s.io \"\" is invalid: force: Forbidden: may not be specified for non-apply patch","reason":"Invalid","details":{"group":"meta.k8s.io","kind":"PatchOptions","causes":[{"reason":"FieldValueForbidden","message":"Forbidden: may not be specified for non-apply patch","field":"force"}]},"code":422}
    2020-05-25 11:13:11.817  INFO 1 --- [nio-8080-exec-1] okhttp3.OkHttpClient                     : <-- END HTTP (392-byte body)

Further, in the examples linked above I see that there is the notion of operation (defined by op field). What are the available operations (link to the docs would help)?

P.S there is a [SO question](https://stackoverflow.com/questions/61998762/kubernetes-java-api-client-change-pods-labels) regarding this if someone wants to answer there.

**Comment by yue9944882** [comment]
hi @FlorinAlexandru , your code looks good except that:

coreV1Api.patchNamespacedPod(pod.getMetadata().getName(), namespace, patch, null, null, null, ~~false~~null); // no need for explict "force" on patch operation

**Comment by FlorinAlexandru** [comment]
@yue9944882 Changing the code as you recommended worked. I was able to change the labels on the pods. I still have  more questions thou:

1. In the above code I added a new label. If I want to change the value of an existing label it's enough to set the new value of an existing key? 

2. What are the  available operation for the _op_ field? I searched in the docs  but I didn't find anything related to this(a link to docs would be great).
